rest/doc_api.go

```diff
@@ -320,6 +330,57 @@ func (h *handler) handlePutDoc() error {
 	return nil
 }
 
+func (h *handler) handlePutDocReplicator2(docid string) (newRev string, doc *db.Document, err error) {
+	bodyBytes, err := h.readBody()
+	if err != nil {
+		return "", nil, err
+	}
+	if bodyBytes == nil || len(bodyBytes) == 0 {
+		return "", nil, base.ErrEmptyDocument
+	}
+
+	newDoc := &db.Document{
+		ID: docid,
+	}
+	newDoc.UpdateBodyBytes(bodyBytes)
+
+	var parentRev string
+	if oldRev := h.getQuery("rev"); oldRev != "" {
+		parentRev = oldRev
+	} else if ifMatch := h.rq.Header.Get("If-Match"); ifMatch != "" {
+		parentRev = ifMatch
+	}
+
```

Comment:
I guess there's still a bit of a semantic gap between blip and a replicator2 PUT.  

Blip:
 - rev == new revision id
 - history (optional) == revision ancestor(s)

PUT replicator2:
 - rev == parent revision

I think that's probably reasonable for now (for ease of compatibility with the existing performance tests), but we'll probably want to think about how to support the concept of 'history+new rev' via REST in a future enhancement.  It's possible we could just toggle to that based on new_edits=false, like a regular PUT.